util/reggen/uvm_reg.sv.tpl

```diff
@@ -135,6 +136,7 @@ package ${block.name}_ral_pkg;
     function new(string name = "${gen_dv.bcname(block)}",
                  int    has_coverage = UVM_NO_COVERAGE);
       super.new(name, has_coverage);
+      add_enable_reg();
```

Comment:
Actually, you would be calling this in the build function and pass the actual lock reg as arg.

Reply:
oh I see your point, that is right, calling here before build is not going to add any locked regs. Thank you for pointing this out Sri.